javalanglib/src/main/scala/java/lang/ClassValue.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.lang
+
+import java.util.HashMap
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import scala.scalajs.runtime.linkingInfo
+
+import Utils._
+
+abstract class ClassValue[T] protected () {
+  private val jsMap: js.Map[Class[_], T] = {
+    if (linkingInfo.assumingES6 || js.typeOf(js.Dynamic.global.Map) != "undefined")
```

Comment:
Use `js.constructorOf[js.Map]` like in the previous commit?

Reply:
That didn't work, actually. It threw a `ReferenceError`.